app/web/features/auth/useAuthStore.ts

```diff
@@ -161,12 +135,47 @@ export default function useAuthStore() {
         }
         setLoading(false);
       },
+      async checkAuthStatus() {
+        try {
+          const res = await service.user.getAuthState();
+          if (res.loggedIn) {
+            console.log(
+              "We thought we were not logged in but an API call shows we were."
+            );
+            const auth = res.authRes!;
+            setUserId(auth.userId);
+            Sentry.setUser({ id: auth.userId.toString() });
+
+            //this must come after setting the userId, because calling setQueryData
+            //will also cause that query to be background fetched, and it needs
+            //userId to be set.
+            setJailed(auth.jailed);
+            setAuthenticated(true);
+          }
+        } catch (e) {
+          Sentry.captureException(e, {
+            tags: {
+              component: "auth/useAuthStore",
+              action: "checkAuthStatus",
+            },
+          });
+          setError(isGrpcError(e) ? e.message : fatalErrorMessage.current);
+        }
+      },
     }),
     //note: there should be no dependenices on the state or t, or
     //some useEffects will break. Eg. the token login in Login.tsx
     [setAuthenticated, setJailed, setUserId, setFlowState, queryClient]
   );
 
+  // useEffect(() => {
+  //   // if we aren't logged in and are otherwise idle, but auth state changed, check if the cookie is set in the bg
+  //   if (typeof window !== "undefined" && !authenticated && !loading && !error && !checkedAuthStatus) {
+  //     setCheckedAuthStatus(true);
+  //     authActions.checkAuthStatus();
+  //   }
+  // }, [authenticated, checkedAuthStatus, setCheckedAuthStatus, authActions]);
+
```

Comment:
Remove?

Reply:
Moved to https://github.com/Couchers-org/couchers/pull/4693